ert_data/obs_loader.py

```diff
@@ -0,0 +1,95 @@
+import pandas as pd
+
+
+def data_loader_factory(observation_type):
+    """
+    Currently, the methods returned by this factory differ. They should not.
+    TODO: Remove discrepancies between returned methods.
+        See https://github.com/equinor/libres/issues/808
+    """
+    if observation_type == "GEN_OBS":
+        return load_general_data
+    elif observation_type == "SUMMARY_OBS":
+        return load_summary_data
+    elif observation_type == "BLOCK_OBS":
+        return load_block_data
+    else:
+        raise TypeError("Unknown observation type: {}".format(observation_type))
+
+
+def load_general_data(facade, observation_key, case_name):
+    observations = facade.get_observations()
+    data = pd.DataFrame()
+    if observation_key not in observations:
+        return data
+
+    obs_vector = facade.get_observations()[observation_key]
+
+    for time_step in obs_vector.getStepList().asList():
+
+        # Observations and its standard deviation are a subset of the simulation data.
+        # The index_list refers to indices in the simulation data. In order to
+        # join these data in a DataFrame, pandas inserts the obs/std
+        # data into the columns representing said indices, before appending
+        # the simulation data.
+        # You then get something like:
+        #      0   1   2
+        # OBS  NaN NaN 42
+        # STD  NaN NaN 4.2
+        #   0  1.2 2.3 7.5
+        node = obs_vector.getNode(time_step)
+        index_list = [node.getIndex(nr) for nr in range(len(node))]
+
+        data = (
```

Comment:
maybe there is a reason to write it like this - but why not just put `get_data_points` and `get_std()` as two columns directly?